R/zzz.R

```diff
@@ -17,3 +17,4 @@
 ## }
  
 variogram <- function(object, ...) UseMethod("variogram")
+units.default <- function(x, ...) base::units.difftime(x)
```

Comment:
Why?

Reply:
@edzer Thanks for checking! My data is a `STIDF` with `zoo::yearmon` as its time information. When I calculate the semi-variogram, I keep getting this error:

> Error in UseMethod("units") : no applicable method for 'units' applied to an object of class "c('double', 'numeric')"

I found the error results from the following lines of `variogramST.STIDF`:

https://github.com/r-spatial/gstat/blob/8335d8e2fb1a16b02d256148d502a7edcff9142a/R/variogramST.R#L191-L192

`diff()` on a `yearmon` object returns a pure numeric vector with no unit, and `units()` is a generic function without implementation on it.

The same problem arises when I do spatio-temporal interpolations with `krigeST()`.

https://github.com/r-spatial/gstat/blob/8335d8e2fb1a16b02d256148d502a7edcff9142a/R/krigeST.R#L97

`data@time` and `newdata@time` are both `yearmon`, and `outer` on them gets a numeric matrix without unit, which make `units()` break down again.

---

My workaround is to define

```r
units.default <- function(x, ...) base::units.difftime(x)
```

so that the time difference of `yearmon` will be dispatched to `base::units.difftime()`, and returns `NULL` instead of errors. Maybe there are other preferable solutions.

Reply:
I see, but I'm afraid it is the wrong fix; below is another attempt to fix this problem.

Reply:
Thank you for taking the time to help! Maybe the following part from `krigeST` involving `units()` also need to be revised.

https://github.com/r-spatial/gstat/blob/8335d8e2fb1a16b02d256148d502a7edcff9142a/R/krigeST.R#L97

(this does not affect my current work.) You may consider my initial suggestion about `variogramST.STIDF` (#135) or be free close it. 

All the best,